flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/DataSetSessionWindowAggregateProcessor.scala

```diff
@@ -59,7 +61,9 @@ class DataSetSessionWindowAggregateReduceGroupFunction(
     finalRowWindowEndPos: Option[Int],
     gap:Long,
     isInputCombined: Boolean)
-  extends RichGroupReduceFunction[Row, Row] {
+  extends AbstractRichFunction
+  with MapPartitionFunction[Row, Row]
```

Comment:
I like the idea of implementing a joint `MapPartition` and `GroupReduce` function. However, I think it is not necessary for the final aggregation. We can also call `DataSet.reduceGroup()` which will do the same as `DataSet.mapPartition().parallelism(1)`.